plugins/Goals/API.php

```diff
@@ -53,6 +53,26 @@ class API extends \Piwik\Plugin\API
     const AVG_PRICE_VIEWED = 'avg_price_viewed';
 
     /**
+     * Return a single goal.
+     *
+     * @param int $idSite
+     * @param int $idGoal
+     * @return array An array of goal attributes.
+     */
+    public function getGoal($idSite, $idGoal)
+    {
+        Piwik::checkUserHasViewAccess($idSite);
+        
+        $goals = $this->getGoals($idSite);
```

Comment:
using getGoals as those goals are cached and there was no DB method yet to fetch a single goal. I can add the method to fetch only a single goal in DB and then return it directly instead of iterating over goals